Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableSystemComponent.cpp

```diff
@@ -20,11 +21,27 @@ namespace AzFramework
     {
         Spawnable::Reflect(context);
         SpawnableMetaData::Reflect(context);
+        EntitySpawnTicket::Reflect(context);
 
         if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context); serializeContext != nullptr)
         {
             serializeContext->Class<SpawnableSystemComponent, AZ::Component>();
             serializeContext->RegisterGenericType<AZ::Data::Asset<Spawnable>>();
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<EntitySpawnTicket>("EntitySpawnTicket", "An object used to spawn prefabs with SpawnableEntitiesInterface.");
+            }
+        }
+
+        if (auto behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext->Class<EntitySpawnTicket>("EntitySpawnTicket")
+                ->Constructor()
```

Comment:
Should we have a method to release this ticket so that we don't have to depend on the destructor being called via garbage collection? I'm not too familiar with the LUA bindings and how reliably objects are destroyed.

Reply:
I think that we should not rely on destructor nor force user to call some additional ticket.Destroy() function. We already have implicit Despawn to do this. But maybe @AMZN-koppersr can answer this better.

Reply:
@AMZN-koppersr "If one is not provided, a new ticket is automatically created." - but ticket requires a spawnable asset. Where would that come from?